OpenQA/Isotovideo/NeedleDownloader.pm

```diff
@@ -124,6 +124,10 @@ sub download ($self) { $self->_download_file($_) for (@{$self->files_to_download
 # downloads missing needles considering $new_needles
 # (see t/21-needle-downloader.t for an example of $new_needles)
 sub download_missing_needles ($self, $new_needles) {
+    if ($new_needles && $bmwqemu::vars{SYNC_ASSETS_HOOK}) {
+        bmwqemu::diag("Running SYNC_ASSETS_HOOK");
+        system($bmwqemu::vars{SYNC_ASSETS_HOOK});
+    }
```

Comment:
If a hook script is specified the "normal" needle download should probably be skipped as it would not work anyway. Maybe the skipping could depend on the exit code of the hook script.

When skipping the download you could maybe also avoid the moving via `Test::MockObject::Extends` in the test code.

Reply:
I've done it by some non-default exit code (32) as needles download might still be useful for example with alternative needles dir (openqa-clone-custom-git-refspec?) or when just "factory" dir (but not "tests") is covered by this sync hook.